project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module

```diff
@@ -106,3 +106,47 @@ function _c4m_user_subscribe_entity_notifications_disabled($entity_id, $flag_suf
 
   return FALSE;
 }
+
+/**
+ * Implements hook_og_membership_type_insert().
+ *
+ * Flags the group subscription flag, so the user is subscribed to the group.
+ */
+function c4m_user_subscribe_og_membership_insert(OgMembership $membership) {
+  c4m_user_subscribe_to_group($membership);
+}
+
+/**
+ * Implements hook_og_membership_type_insert().
+ *
+ * Flags the group subscription flag, so the user is subscribed to the group.
+ */
+function c4m_user_subscribe_og_membership_update(OgMembership $membership) {
+  c4m_user_subscribe_to_group($membership);
+}
+
+/**
+ * Implements hook_og_membership_type_delete().
+ *
+ * Unflags the group subscription flag, so the user is no longer subscribed to
+ * the group.
+ */
+function c4m_user_subscribe_og_membership_delete(OgMembership $membership) {
+  c4m_user_subscribe_to_group($membership, 'unflag');
+}
+
+/**
+ * Updates the subscription flag to a group from the membership.
+ *
+ * @param object $membership
+ *   OgMembership object containing the information of the group and the state.
+ * @param string $event
+ *   Actions to update the subscription flag.
+ *   Possible values: "flag", "unflag".
+ */
+function c4m_user_subscribe_to_group($membership, $event = 'flag') {
+  if ($membership->state == OG_STATE_ACTIVE) {
```

Comment:
I haven't tested it, but with that single check we are preventing a larger piece of code being called. Do you want me to add that check only to the insert/update operations?